woocommerce/google-listings-and-ads
# Connect Programs filter to data source.

### Changes proposed in this Pull Request:

Create `ProgramsReportFilters` to use customized `ReportsFilter`, that will use Date, Programs, and 
Compare filters consistently with other wc-admin analytics.
Provide custom autocompleter for programs.

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/242.

:warning: Requires wc-admin fixes
 - `CompareFilter` does not forward `autocompleter` setting to the `Search`
https://github.com/woocommerce/woocommerce-admin/issues/6890
 - FilterPicker does not forward all props to Search component
https://github.com/woocommerce/woocommerce-admin/issues/6062

### Screenshots:

![Animation of filter picker being used and populated with real programs data](https://user-images.githubusercontent.com/17435/116718350-2a258a00-a9da-11eb-990d-7c204950a9ae.gif)

### Review hints:

Changes to our components are in the first two commits https://github.com/woocommerce/google-listings-and-ads/pull/537/files/16728a6465480a81cc5a599da67fe1fcbebfe931

The others are importing unreleased wc-admin fixes: https://github.com/woocommerce/google-listings-and-ads/pull/537/files/16728a6465480a81cc5a599da67fe1fcbebfe931..feaff0901cb3997a6bda7e99773741e205756208

### Detailed test instructions:

1. Go to Programs report page: https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms
2. Use date and program filter (all, single, compare)
3. Check that params in the address bar URL updates accordingly, and UI looks correct.
4. Check that the autocomplete is populated with actual programs.


### To be done in a separate PR:

1. Fix initial state. When you refresh the page or visit URL for the first time with programs chosen, the labels are not shown. This seems to ba another wc-admin issue, as `CompareFilter` & `FilterPicker` seems to ignore changing `getLabels` function (it updates labels, only when params changes)

	![Screencast that shows missing labels](https://user-images.githubusercontent.com/17435/116718858-b768de80-a9da-11eb-8134-ebadab26c6f9.gif)

2. When "Freee Listings" are chosen as a "single program", their label is not rendered. Probably because `FilterPicker` considers the selected option with `id==0` the lack of selection.
![Screencast showing the lack of label](https://user-images.githubusercontent.com/17435/116720453-7f629b00-a9dc-11eb-9116-274d95c6211b.gif)


### Changelog Note:
